frontend/components/shared/FormWrapper.vue

```diff
@@ -0,0 +1,146 @@
+<script lang="ts" setup>
+import { computed } from "vue";
+import { BSButton } from "cdh-vue-lib";
+import FormStepper, { type FormStepperConfig } from "./FormStepper.vue";
+import { type GetFormQuery } from "~/generated/gql/graphql";
+
+export type TopLevelForm = NonNullable<GetFormQuery["form"]>;
+export type Subform = NonNullable<TopLevelForm["subforms"][number]>;
+export type NestedSubform = NonNullable<Subform["subforms"][number]>;
+
+export type CombinedForm = TopLevelForm | Subform | NestedSubform;
+
+interface Props {
+    form: NonNullable<GetFormQuery["form"]>;
+    currentFormSlug: string;
+}
+
+const props = defineProps<Props>();
+
+const formStepperConfig = computed<FormStepperConfig>(() => {
+    const selectedFormSlug = props.currentFormSlug;
+
+    const topLevelForm = props.form;
+
+    return {
+        titleNl: topLevelForm.nameNl ?? "",
+        titleEn: topLevelForm.nameEn ?? "",
+        steps: topLevelForm.subforms.map((form) => ({
+            slug: form.slug,
+            labelNl: form.nameNl ?? "",
+            labelEn: form.nameEn ?? "",
+            completed: false,
+            active: form.slug === selectedFormSlug,
+            disabled: false,
+            children: form.subforms.map((subform) => ({
+                slug: subform.slug,
+                labelNl: subform.nameNl ?? "",
+                labelEn: subform.nameEn ?? "",
+                completed: false,
+                active: subform.slug === selectedFormSlug,
+                disabled: false,
+                // Let's only go 2 levels deep for now.
+                children: [],
+            })),
+        })),
+    };
+});
+
+const allForms = computed(() => getAllForms(props.form));
+const selectedForm = computed(() => {
+    const forms = allForms.value;
+    if (forms.length <= 0) {
+        return null;
+    }
+    return forms.find(({ slug }) => slug === props.currentFormSlug) ?? null;
+});
+
+function getAllForms(form: CombinedForm) {
+    const collectedForms = [form];
+    if ("subforms" in form) {
+        form.subforms.forEach((subform) => {
+            collectedForms.push(...getAllForms(subform));
+        });
+    }
+    return collectedForms;
+}
```

Comment:
Maybe this can become something like this:
```suggestion
function getAllForms(form: CombinedForm) {
    const collectedForms: CombinedForm[] = [];
    if ("subforms" in form && Array.isArray(form.subforms)) {
        form.subforms.forEach((subform) => {
            collectedForms.push(subform, ...getAllForms(subform));
        });
    }
    return collectedForms;
}
```
So that the top level form gets excluded from the normal forms again :)